### PR TITLE
chore: feature reports: build errors now have a logs address

### PR DESCRIPTION
**Pull request by AurelienGasser**
Companion PRs:
- https://github.com/Substra/substra-backend/pull/483 (main PR)
- https://github.com/Substra/orchestrator/pull/38 (this PR)
- https://github.com/Substra/substra/pull/295
- https://github.com/Substra/substra-tests/pull/218
- https://github.com/Substra/substra-frontend/pull/116
- https://github.com/Substra/substra-documentation/pull/184

--- 

## Description

Previously, only _execution_ errors had logs. Now, build errors have logs too.

## How has this been tested?

Tested up and down migrations on pre-existing data.

## Checklist

- [x] [changelog](../CHANGELOG.md) was updated with notable changes
- [x] documentation was updated

### CODE REVIEW

File: lib/asset/failure_report_validation.go [hunk 1]
```diff
@@ -3,13 +3,14 @@ package asset
 import (
 	validation "github.com/go-ozzo/ozzo-validation/v4"
 	"github.com/go-ozzo/ozzo-validation/v4/is"
+	"golang.org/x/exp/slices"
```
Comment (mblottiere) [inline lib/asset/failure_report_validation.go:6]: Please do not use experimental modules, they have no stability guarantees.

File: lib/asset/failure_report_validation.go [hunk 2]
```diff
@@ -3,13 +3,14 @@ package asset
...
 )
 
 // Validate returns an error if the new FailureReport object is not valid.
 func (f *NewFailureReport) Validate() error {
 	return validation.ValidateStruct(f,
 		validation.Field(&f.ComputeTaskKey, validation.Required, is.UUID),
 		validation.Field(&f.ErrorType, validation.In(ErrorType_ERROR_TYPE_BUILD, ErrorType_ERROR_TYPE_EXECUTION, ErrorType_ERROR_TYPE_INTERNAL)),
-		validation.Field(&f.LogsAddress, validation.When(f.ErrorType == ErrorType_ERROR_TYPE_EXECUTION, validation.Required).Else(validation.Nil)),
+		validation.Field(&f.LogsAddress, validation.When(slices.Contains([]ErrorType{ErrorType_ERROR_TYPE_EXECUTION, ErrorType_ERROR_TYPE_BUILD}, f.ErrorType), validation.Required).Else(validation.Nil)),
```
Comment (mblottiere) [inline lib/asset/failure_report_validation.go:14]: We already have a method for that check: https://github.com/Substra/orchestrator/blob/d5ec8c67c63bbbeeafb797b4808514976cf05fb1/utils/slice.go#L10

File: lib/asset/failure_report_validation.go (hunk 1 above)
Comment (AurelienGasser) [reply lib/asset/failure_report_validation.go]: done

File: lib/asset/failure_report_validation.go (hunk 2 above)
Comment (AurelienGasser) [reply lib/asset/failure_report_validation.go]: done

**Review by mblottiere — APPROVED** [review]

**Review by thibowk — APPROVED** [review]

**Review by AlexandrePicosson — APPROVED** [review]

**Pull request merged**